### ISSUE TITLE
Add `getNetworkFromAddress` method

### ISSUE THREAD

**Issue by janniks** [stx-labs/stacks.js#1485]
> It might make sense to add network mode that determines how addresses are encoded.
> This is used in the explorer
> Like a getNetworkFromAddress? It would result in Testnet for Devnet/Mocknet addresses, since there's no difference. Is that acceptable?

Issue brought up by @friedger in #1470

**Comment by ri-dev-git** [comment]
Hi @janniks! can you explain what needs to be done with an example?

**Comment by janniks** [comment]
Hi, would you like to contribute? ✨

_In short I would:_
Add a function `networkFromAddress(address: string)` to the transactions package (packages/transactions/src/utils.ts) which decodes the given address (using c32addressDecode) and compares the version (of the address) to mainnet vs testnet, then returns a new instance of that network.
Add some simple test cases (to utils.test.ts) which cover an address for each network (devnet/mocknet also just use testnet address versions)

### pseudo example

```ts
const network = networkFromAddress("SP8931298312");
    //^? instance of StacksMainnet()
```

> Although maybe it's better to add other helpers like `isMainnet(address)` which could be combined easily. 🤷‍♂

**Comment by friedger** [comment]
How can you differentiate between devnet, testnet, etc when they all use the same address

**Comment by janniks** [comment]
Exactly, that doesn’t work and why I believe some other method might be better suited. 😕
The network would have the correct version, but shouldn’t be used for broadcast etc.

**Comment by ri-dev-git** [comment]
hey @janniks! i have implemented it as something like this:
```
export const networkFromAddress=(address: string): {network:string,address:string} =>{
  try {
    const decode=c32addressDecode(address);
    if(isTestnetAddress(decode)){
      return {
        network:"Testnet",
        address: address
      }
    }else{
      return{
        network:"Testnet",
        address: address
      }
    }
  }
  catch(e){
    return e
  }
}
```

**Comment by janniks** [comment]
Hi, maybe we should instead add a new method like `isMainnet(address: string)` which can easily be combined with others -- let's hold off on the actual implementation and gather some feedback? 🤔

_e.g. easy use of `isMainnet` to create new helpers_
```ts
const networkFromAddress = (address: string) => (isMainnet(address) ? new StacksMainnet() : new StacksTestnet());
```

**Comment by ri-dev-git** [comment]
Yeah ok works!
